### DISCUSSION TITLE
Fill missing data

### DISCUSSION THREAD

**Discussion by rakhmatsuryahadi** [teamalgoritma/community#905] [Q&A] [answered]
Saya ingin menanyakan terkait dengan data yang hilang dimana kolom data tersebut merupakan unit price. apakah dengan bfill atau ffill dapat membantu secara valid pengisian data yang kosong tersebut? saya berpandangan bahwa unit price tidak bisa kita isi dengan function karena hal tersebut berdampak terhadap kalkulasi bisnis. namun jika terlalu banyak row yg missing tentunya akan menjadi PR sekali untuk mengisinya secara manual. mohon pencerahannya

**Comment by adizaen** [comment, marked as answer]
Dear @rakhmatsuryahadi 

Terima kasih atas pertanyaannya Pak. 
Untuk treatment missing values sebenarnya ada beberapa cara umum yang bisa digunakan Pak selain `bfill` dan `ffill`. Antara lain:

1. Hapus baris atau kolom: Menggunakan metode `dropna()`
2. Replace NA dengan nilai `mean`, `median`, dll
3. Tetap mempertahankan data kita

Sebenarnya jika total missing value yang Bapak miliki < 10% total data, datanya bisa dihapus Pak. Tidak masalah. Tetapi jika datanya tidak ingin dihapus untuk menghindari hilangnya informasi, ada proses imputasi untuk mengisi value tersebut Pak. Bapak bisa melakukan imputasi terhadap data yang mengandung missing value, menggunakan metode `.fillna()`. Untuk value yang diisikan bisa berupa `mean` atau `median` dari kolom `unit price`.

Semoga membantu Pak.

Best regards,
**Team Algoritma**

**Comment by saskia-dwi-ulfah** [comment]
Dear @rakhmatsuryahadi 

Terima kasih telah bertanya melalui Github discussion Algoritma. Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

Best regards,
Team Algoritma
